yapapi/runner/events.py

```diff
@@ -0,0 +1,155 @@
+"""Representing and logging events in Golem computation."""
+from enum import Enum, auto
+from typing import Any, Optional, TypeVar, Union
+import sys
+
+from typing_extensions import Protocol
+
+if sys.version_info >= (3, 8):
+    from typing import get_args as get_type_args
+else:
+    from typing_extensions import get_args as get_type_args  # type: ignore
+
+
+class _EventBase:
```

Comment:
Why not just _Event?

Reply:
I couldn't come up with another a good name for this mixin class that just provides `__str__()` but with an implementation that requires that the self object is an event type. `_PrintableEventType`?  `_EventWithStr`?.

Reply:
Renamed to `_EventStrMixin`!